monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -0,0 +1,370 @@
+
+    Args:
+        algo: a dictionary that outlines the algorithm to use.
+        data_stats_filename: the path to the data stats file (generated by DataAnalyzer)
+        data_src_cfg_name: the path to the data source file, which contains the data list
+
+    .. code-block:: bash
+
+        python -m monai.apps.auto3dseg BundleGen generate --data_stats_filename="../algorithms/data_stats.yaml"
+    """
+
+    def __init__(self, algos=None, data_stats_filename=None, data_src_cfg_name=None):
+        self.algos = []
+        if algos is None:
+            # trigger the download process
+            sys.path.insert(0, "..")
+            algo_compressed_file = os.path.join("..", "algo_templates.tar")
+            download_and_extract(algo_zip, algo_compressed_file, "..", md5)
+            algos = copy(default_algos)
+        if isinstance(algos, dict):
+            for algo_name, algo_params in algos.items():
+                self.algos.append(ConfigParser(algo_params).get_parsed_content())
+                self.algos[-1].name = algo_name
+        else:
+            self.algos = ensure_tuple(algos)
+
+        self.data_stats_filename = data_stats_filename
+        self.data_src_cfg_filename = data_src_cfg_name
+        self.history: List[Dict] = []
+
+    def set_data_stats(self, data_stats_filename: str):  # type: ignore
+        """
+        Set the data stats filename
+
+        Args:
+            data_stats_filename: filename of datastats
+        """
+        self.data_stats_filename = data_stats_filename
+
+    def get_data_stats(self):
+        """Get the filename of the data stats"""
+        return self.data_stats_filename
+
+    def set_data_src(self, data_src_cfg_filename):
+        """
+        Set the data source filename
+
+        Args:
+            data_src_cfg_filename: filename of data_source file
+        """
+        self.data_src_cfg_filename = data_src_cfg_filename
+
+    def get_data_src(self, fold_idx=0):
+        """Get the data source filename"""
+        return self.data_src_cfg_filename
+
+    def get_history(self, *args, **kwargs) -> List:
+        """get the history of the bundleAlgo object with their names/identifiers"""
+        return self.history
+
+    def generate(self, output_folder=".", fold_idx: Union[int, Sequence[int]] = 0):
+        """
+        Generate the bundle scripts/configs for each bundleAlgo
+
+        Args:
+            output_folder: the output folder to save each algorithm.
+            fold_idx: an index to append to the name of the algorithm to
+        """
+        for algo in self.algos:
+            for f_id in ensure_tuple(fold_idx):
+                data_stats = self.get_data_stats()
+                data_src_cfg = self.get_data_src(fold_idx)
+                gen_algo = deepcopy(algo)
+                gen_algo.set_data_stats(data_stats)
+                gen_algo.set_data_source(data_src_cfg)
+                name = f"{gen_algo.name}_{f_id}"
```

Comment:
might need a UUID here to have a unique bundle name?

Reply:
@Nic-Ma and I happened to discuss UUID on another day. Looks like a single f_id satisfy the need of multi-fold, but when HPO comes in, better identifier is needed, and it would be best if the name can tell the experiment details.